### PR TITLE
feat: use reshuffleByTimestamp + added BasciClient + tests

### PR DESCRIPTION
**Pull request by gpuente**
## Description:

- fix sort problem for operations after merge conflict

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -47,3 +49,98 @@ export function buildOperations(
...
+    return {
+        document: newDocument,
+        operation: {
+            ...operation,
+            index: index ?? operation.index
+        } as Operation<NOOPAction & Action>
+    };
+}
+
+export class BasicClient {
```
Comment (gpuente) [inline test/utils.ts:77]: now we can use this client to simulate how connect sends operations and sync with switchboard

File: test/server/processOperations.test.ts [hunk 2]
```diff
@@ -461,4 +461,242 @@ describe('processOperations', () => {
...
+
+        client1.dispatchDocumentAction(actions.setModelId({ id: '4' }));
+        pushOpreationResult = await client1.pushOperationsToServer();
+        expect(pushOpreationResult.status).toBe('SUCCESS');
+
+        // Check the final state of the document
+        const finalDocument = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
```
Comment (gpuente) [inline test/server/processOperations.test.ts:622]: here there's an example of how we can use this new client to simulate different scenarios

File: test/server/processOperations.test.ts [hunk 3]
```diff
@@ -461,4 +461,242 @@ describe('processOperations', () => {
...
+                input: { id: '4' },
+                index: 9,
+                skip: 0
+            }
+        ]);
+    });
+
+    it('should resolve conlficts suign the right order for merge oprations (simulate clients conflict)', async () => {
+        const initialDocument = await buildFile();
+        let pushOpreationResult: IOperationResult;
```
Comment (acaldas) [inline test/server/processOperations.test.ts:584]: typo: pushOperation

File: test/server/processOperations.test.ts [hunk 4]
```diff
@@ -461,4 +461,242 @@ describe('processOperations', () => {
...
+            {
+                type: 'SET_MODEL_ID',
+                input: { id: '4' },
+                index: 9,
+                skip: 0
+            }
+        ]);
+    });
+
+    it('should resolve conlficts suign the right order for merge oprations (simulate clients conflict)', async () => {
```
Comment (acaldas) [inline test/server/processOperations.test.ts:582]: I can feel how tired you were when doing this ahah

typo: should resolve conflicts using the right order for merge operations

**Review by acaldas — APPROVED** [review]

File: test/server/processOperations.test.ts (hunk 4 above)
Comment (gpuente) [reply test/server/processOperations.test.ts]: added fix here: [8cb126d](https://github.com/powerhouse-inc/document-drive/pull/129/commits/8cb126d241bb8db0809b763990a0a45f2050c627)

File: test/server/processOperations.test.ts (hunk 3 above)
Comment (gpuente) [reply test/server/processOperations.test.ts]: added fix here: [9452c5c](https://github.com/powerhouse-inc/document-drive/pull/129/commits/9452c5ce853bdb94856ba35d074fb2934d2cbd73)

**Pull request merged**